packages/filebrowser/src/browser.ts

```diff
@@ -180,6 +220,49 @@ export class FileBrowser extends SidePanel {
     }
   }
 
+  /**
+   * Whether to show a text box to filter files by name.
+   */
+  get showFileFilter(): boolean {
+    return this._showFileFilter;
+  }
+
+  set showFileFilter(value: boolean) {
+    // If the old value was true and the new value is false, clear the filter
+    const oldValue = this.showFileFilter;
+    if (oldValue && !value) {
+      // Clear the search box input
+      (this.filterToolbar.layout as PanelLayout).widgets.forEach(widget => {
+        if (
+          widget.node.getAttribute('data-jp-item-name') === 'fileNameSearcher'
+        ) {
+          // Find the search element's "clear" button and click it
+          widget.node.children[0].shadowRoot?.childNodes.forEach(childNode => {
+            if (childNode.nodeType === Node.ELEMENT_NODE) {
+              const clearButtonElements = (
+                childNode as HTMLElement
+              ).getElementsByClassName('clear-button');
+              if (clearButtonElements.length > 0) {
+                (clearButtonElements[0] as HTMLButtonElement).click();
```

Comment:
This looks pretty involved. @fcollonval is there a simpler way of cleaning the input?